### PR TITLE
WIP: Transition repo to typescript

### PR DESCRIPTION
**Pull request by birkskyum**
Replace require() with import/export
Replace tape with jest
Replace flow-types with typescript in react components

### CODE REVIEW

File: docs/components/api/examples.tsx [hunk 1]
```diff
@@ -22,13 +21,13 @@ export default class Examples extends React.Component {
...
-            PropTypes.shape({
-                caption: PropTypes.string,
-                description: PropTypes.string.isRequired
-            })
-        ).isRequired
-    }).isRequired
+type ExamplesProps = {
+    section: any;
+    headingLevel?: number;
+    // {
```
Comment (HarelM) [inline docs/components/api/examples.tsx:27]: is this deliberate?

File: docs/components/api/github.tsx [hunk 2]
```diff
@@ -21,13 +20,7 @@ export default class GitHub extends React.Component {
...
-    section: PropTypes.shape({
-        context: PropTypes.shape({
-            github: PropTypes.shape({
-                url: PropTypes.string,
-                path: PropTypes.string
-            })
-        })
-    })
+type GitHubProps = {
+    section?: any,
```
Comment (HarelM) [inline docs/components/api/github.tsx:24]: This can be better typed according to the removed code I think.

**Comment by HarelM** [comment]
I've seen some places where the types were "reduced". Please improve that and let me know when this is ready as there are a lot of files here and I would like to go over them as little as possible...

**Comment by birkskyum** [comment]
@HarelM , of course. It's mainly these sections that you noticed. They were troublesome so I temporarily disable the types, but they'll come back now.

**Comment by HarelM** [comment]
Sure.
Ping me when CI passes and I'll review the changes.
I saw that you removed the submodule as part of this commit, great!!
Keep up the good work!

**Comment by birkskyum** [comment]
Right now the package.json command "generate-batfish-input" makes a root folder "batfish-input" that has all the javascript, generated images etc that should be needed for batfish, but for some reason it isn't satisfied with the configuration file I point it towards - so the missing link is how to make batfish run our js input.

**Pull request closed without merging**